apps/dash-oil-and-gas/app.py

```diff
@@ -0,0 +1,715 @@
+                                html.Div(
+                                    [html.H6(id="gasText"), html.P("Gas")],
+                                    id="gas",
+                                    className="mini_container",
+                                ),
+                                html.Div(
+                                    [html.H6(id="oilText"), html.P("Oil")],
+                                    id="oil",
+                                    className="mini_container",
+                                ),
+                                html.Div(
+                                    [html.H6(id="waterText"), html.P("Water")],
+                                    id="water",
+                                    className="mini_container",
+                                ),
+                            ],
+                            id="info-container",
+                            className="row container-display",
+                        ),
+                        html.Div(
+                            [dcc.Graph(id="count_graph")],
+                            id="countGraphContainer",
+                            className="pretty_container",
+                        ),
+                    ],
+                    id="right-column",
+                    className="eight columns",
+                ),
+            ],
+            className="row flex-display",
+        ),
+        html.Div(
+            [
+                html.Div(
+                    [dcc.Graph(id="main_graph")],
+                    className="pretty_container seven columns",
+                ),
+                html.Div(
+                    [dcc.Graph(id="individual_graph")],
+                    className="pretty_container five columns",
+                ),
+            ],
+            className="row flex-display",
+        ),
+        html.Div(
+            [
+                html.Div(
+                    [dcc.Graph(id="pie_graph")],
+                    className="pretty_container seven columns",
+                ),
+                html.Div(
+                    [dcc.Graph(id="aggregate_graph")],
+                    className="pretty_container five columns",
+                ),
+            ],
+            className="row flex-display",
+        ),
+    ],
+    id="mainContainer",
+    style={"display": "flex", "flex-direction": "column"},
+)
+
+
+# Helper functions
+def human_format(num):
+
+    magnitude = int(math.log(num, 1000))
+    mantissa = str(int(num / (1000 ** magnitude)))
+    return mantissa + ["", "K", "M", "G", "T", "P"][magnitude]
+
+
+def filter_dataframe(df, well_statuses, well_types, year_slider):
+    dff = df[
+        df["Well_Status"].isin(well_statuses)
+        & df["Well_Type"].isin(well_types)
+        & (df["Date_Well_Completed"] > dt.datetime(year_slider[0], 1, 1))
+        & (df["Date_Well_Completed"] < dt.datetime(year_slider[1], 1, 1))
+    ]
+    return dff
+
+
+def fetch_individual(api):
```

Comment:
Just a slight confuse on this function. 
`fetch` naming usually entails you are doing a HTTP request. 
`api` naming is unclear what you are passing through.